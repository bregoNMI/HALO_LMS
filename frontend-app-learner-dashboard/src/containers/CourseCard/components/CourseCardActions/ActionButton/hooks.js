import { useWindowSize, breakpoints } from '@openedx/paragon';

export const useIsCollapsed = () => {
  const { width } = useWindowSize();
  return width < breakpoints.medium.maxWidth && width > breakpoints.small.maxWidth;
};

export default useIsCollapsed;
