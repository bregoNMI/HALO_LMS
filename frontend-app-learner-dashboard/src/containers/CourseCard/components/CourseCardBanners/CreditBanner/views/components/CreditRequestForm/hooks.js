import React from 'react';

export const useCreditRequestFormData = (requestData) => {
  const ref = React.useRef(null);
  React.useEffect(() => {
    if (requestData !== null) {
      ref.current.click();
    }
  }, [requestData]);
  return { ref };
};

export default useCreditRequestFormData;
