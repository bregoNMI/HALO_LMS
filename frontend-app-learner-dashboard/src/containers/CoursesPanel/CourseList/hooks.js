import { useWindowSize, breakpoints } from '@openedx/paragon';

export const useIsCollapsed = () => {
  const { width } = useWindowSize();
  return width < breakpoints.medium.maxWidth;
};

export default useIsCollapsed;
