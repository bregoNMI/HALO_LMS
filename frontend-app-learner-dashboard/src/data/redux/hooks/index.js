export * from './app';
export * from './requests';
