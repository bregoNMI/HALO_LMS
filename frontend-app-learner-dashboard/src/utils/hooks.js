import { useIntl } from '@edx/frontend-platform/i18n';

import dateFormatter from './dateFormatter';

export const useFormatDate = () => {
  const { formatDate } = useIntl();
  return (date) => dateFormatter(formatDate, date);
};

export default {
  useFormatDate,
};
