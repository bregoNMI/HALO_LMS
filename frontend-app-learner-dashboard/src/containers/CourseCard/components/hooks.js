import { reduxHooks } from 'hooks';

export const useActionDisabledState = (cardId) => {
  const { isMasquerading } = reduxHooks.useMasqueradeData();
  const {
    canUpgrade, hasAccess, isAudit, isAuditAccessExpired,
  } = reduxHooks.useCardEnrollmentData(cardId);
  const {
    isEntitlement, isFulfilled, canChange, hasSessions,
  } = reduxHooks.useCardEntitlementData(cardId);

  const { resumeUrl, homeUrl, upgradeUrl } = reduxHooks.useCardCourseRunData(cardId);

  const disableBeginCourse = !homeUrl || (isMasquerading || !hasAccess || (isAudit && isAuditAccessExpired));
  const disableResumeCourse = !resumeUrl || (isMasquerading || !hasAccess || (isAudit && isAuditAccessExpired));
  const disableViewCourse = !hasAccess || (isAudit && isAuditAccessExpired);
  const disableSelectSession = !isEntitlement || isMasquerading || !hasAccess || (!canChange || !hasSessions);
  const disableUpgradeCourse = !upgradeUrl || (isMasquerading && !canUpgrade);

  const disableCourseTitle = (isEntitlement && !isFulfilled) || disableViewCourse;

  return {
    disableBeginCourse,
    disableResumeCourse,
    disableViewCourse,
    disableUpgradeCourse,
    disableSelectSession,
    disableCourseTitle,
  };
};

export default useActionDisabledState;
