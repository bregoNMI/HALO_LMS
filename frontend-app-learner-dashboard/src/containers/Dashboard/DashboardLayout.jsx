import React from 'react';
import PropTypes from 'prop-types';

import { Container, Col, Row } from '@openedx/paragon';

import WidgetFooter from 'containers/WidgetContainers/WidgetFooter';
import hooks from './hooks';

export const columnConfig = {
  courseList: {
    withSidebar: {
      lg: { span: 12, offset: 0 },
      xl: { span: 8, offset: 0 },
    },
    noSidebar: {
      lg: { span: 12, offset: 0 },
      xl: { span: 12, offset: 0 },
    },
  },
  sidebar: {
    lg: { span: 12, offset: 0 },
    xl: { span: 4, offset: 0 },
  },
};

export const DashboardLayout = ({ children, sidebar: Sidebar }) => {
  const {
    isCollapsed,
    sidebarShowing,
    setSidebarShowing,
  } = hooks.useDashboardLayoutData();

  const courseListColumnProps = sidebarShowing
    ? columnConfig.courseList.withSidebar
    : columnConfig.courseList.noSidebar;

  return (
    <Container fluid size="xl">
      <Row>
        <Col {...courseListColumnProps} className="course-list-column">
          {children}
        </Col>
        <Col {...columnConfig.sidebar} className="sidebar-column">
          {!isCollapsed && (<h2 className="course-list-title">&nbsp;</h2>)}
          <Sidebar setSidebarShowing={setSidebarShowing} />
        </Col>
      </Row>
      <Row>
        <Col>
          <WidgetFooter />
        </Col>
      </Row>
    </Container>
  );
};
DashboardLayout.propTypes = {
  children: PropTypes.node.isRequired,
  sidebar: PropTypes.func.isRequired,
};

export default DashboardLayout;
