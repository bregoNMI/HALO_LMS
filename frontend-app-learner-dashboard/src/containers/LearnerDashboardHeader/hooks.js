import React from 'react';
import { useWindowSize, breakpoints } from '@openedx/paragon';
import track from 'tracking';
import { StrictDict } from 'utils';
import { linkNames } from 'tracking/constants';

import * as module from './hooks';

export const state = StrictDict({
  isOpen: (val) => React.useState(val), // eslint-disable-line
});

export const useIsCollapsed = () => {
  const { width } = useWindowSize();
  const isCollapsed = React.useMemo(() => (width <= breakpoints.large.minWidth), [width]);
  return isCollapsed;
};

export const findCoursesNavClicked = (href) => track.findCourses.findCoursesClicked(href, {
  linkName: linkNames.learnerHomeNavExplore,
});

export const findCoursesNavDropdownClicked = (href) => track.findCourses.findCoursesClicked(href, {
  linkName: linkNames.learnerHomeNavDropdownExplore,
});

export const useLearnerDashboardHeaderData = () => {
  const [isOpen, setIsOpen] = module.state.isOpen(false);
  const toggleIsOpen = () => setIsOpen(!isOpen);

  return {
    isOpen,
    toggleIsOpen,
  };
};

export default {
  useIsCollapsed,
  findCoursesNavClicked,
  findCoursesNavDropdownClicked,
  useLearnerDashboardHeaderData,
};
