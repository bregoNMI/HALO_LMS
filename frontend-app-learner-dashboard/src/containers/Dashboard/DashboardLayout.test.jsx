import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';
import { Col, Row } from '@openedx/paragon';

import hooks from './hooks';
import DashboardLayout, { columnConfig } from './DashboardLayout';

jest.mock('./hooks', () => ({
  useDashboardLayoutData: jest.fn(),
}));

const hookProps = {
  isCollapsed: true,
  sidebarShowing: false,
  setSidebarShowing: jest.fn().mockName('hooks.setSidebarShowing'),
};
hooks.useDashboardLayoutData.mockReturnValue(hookProps);

const props = {
  sidebar: jest.fn(() => 'test-sidebar-content'),
};

const children = 'test-children';

let el;
describe('DashboardLayout', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    el = shallow(<DashboardLayout {...props}>{children}</DashboardLayout>);
  });

  const testColumns = () => {
    it('loads courseList and sidebar column layout', () => {
      const columns = el.instance.findByType(Row)[0].findByType(Col);
      Object.keys(columnConfig.sidebar).forEach(size => {
        expect(columns[1].props[size]).toEqual(columnConfig.sidebar[size]);
      });
    });
    it('displays children in first column', () => {
      const columns = el.instance.findByType(Row)[0].findByType(Col);
      expect(columns[0].children).not.toHaveLength(0);
    });
    it('displays sidebar prop in second column', () => {
      const columns = el.instance.findByType(Row)[0].findByType(Col);
      expect(columns[1].findByType(props.sidebar)).toHaveLength(1);
    });
    it('displays a footer in the second row', () => {
      const columns = el.instance.findByType(Row)[1].findByType(Col);
      expect(columns[0].children[0].type).toEqual('WidgetFooter');
    });
  };
  const testSidebarLayout = () => {
    it('displays widthSidebar width for course list column', () => {
      const columns = el.instance.findByType(Row)[0].findByType(Col);
      Object.keys(columnConfig.courseList.withSidebar).forEach(size => {
        expect(columns[0].props[size]).toEqual(columnConfig.courseList.withSidebar[size]);
      });
    });
  };
  const testNoSidebarLayout = () => {
    it('displays noSidebar width for course list column', () => {
      const columns = el.instance.findByType(Row)[0].findByType(Col);
      Object.keys(columnConfig.courseList.noSidebar).forEach(size => {
        expect(columns[0].props[size]).toEqual(columnConfig.courseList.noSidebar[size]);
      });
    });
  };
  const testSnapshot = () => {
    test('snapshot', () => {
      expect(el.snapshot).toMatchSnapshot();
    });
  };
  describe('collapsed', () => {
    describe('sidebar showing', () => {
      beforeEach(() => {
        hooks.useDashboardLayoutData.mockReturnValueOnce({ ...hookProps, sidebarShowing: true });
      });
      testColumns();
      testSnapshot();
      testSidebarLayout();
    });
    describe('sidebar not showing', () => {
      testColumns();
      testSnapshot();
      testNoSidebarLayout();
    });
    it('does not show spacer component above widget sidebar', () => {
      const columns = el.instance.findByType(Col);
      expect(columns[1].findByType('h2').length).toEqual(0);
    });
  });

  describe('not collapsed', () => {
    const testWidgetSpacing = () => {
      it('shows a blank (nbsp) h2 spacer component above widget sidebar', () => {
        const columns = el.instance.findByType(Col);
        // nonbreaking space equivalent
        expect(columns[1].findByType('h2')[0].children[0].el).toEqual('\xA0');
      });
    };
    describe('sidebar showing', () => {
      beforeEach(() => {
        hooks.useDashboardLayoutData.mockReturnValueOnce({
          ...hookProps,
          isCollapsed: false,
          sidebarShowing: true,
        });
      });
      testColumns();
      testSnapshot();
      testSidebarLayout();
      testWidgetSpacing();
    });
    describe('sidebar not showing', () => {
      beforeEach(() => {
        hooks.useDashboardLayoutData.mockReturnValueOnce({ ...hookProps, isCollapsed: false });
      });
      testColumns();
      testSnapshot();
      testNoSidebarLayout();
      testWidgetSpacing();
    });
  });
});
