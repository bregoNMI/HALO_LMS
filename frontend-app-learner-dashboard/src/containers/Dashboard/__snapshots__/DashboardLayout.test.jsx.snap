// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardLayout collapsed sidebar not showing snapshot 1`] = `
<Container
  fluid={true}
  size="xl"
>
  <Row>
    <Col
      className="course-list-column"
      lg={
        Object {
          "offset": 0,
          "span": 12,
        }
      }
      xl={
        Object {
          "offset": 0,
          "span": 12,
        }
      }
    >
      test-children
    </Col>
    <Col
      className="sidebar-column"
      lg={
        Object {
          "offset": 0,
          "span": 12,
        }
      }
      xl={
        Object {
          "offset": 0,
          "span": 4,
        }
      }
    >
      <mockConstructor
        setSidebarShowing={[MockFunction hooks.setSidebarShowing]}
      />
    </Col>
  </Row>
  <Row>
    <Col>
      <WidgetFooter />
    </Col>
  </Row>
</Container>
`;

exports[`DashboardLayout collapsed sidebar showing snapshot 1`] = `
<Container
  fluid={true}
  size="xl"
>
  <Row>
    <Col
      className="course-list-column"
      lg={
        Object {
          "offset": 0,
          "span": 12,
        }
      }
      xl={
        Object {
          "offset": 0,
          "span": 8,
        }
      }
    >
      test-children
    </Col>
    <Col
      className="sidebar-column"
      lg={
        Object {
          "offset": 0,
          "span": 12,
        }
      }
      xl={
        Object {
          "offset": 0,
          "span": 4,
        }
      }
    >
      <mockConstructor
        setSidebarShowing={[MockFunction hooks.setSidebarShowing]}
      />
    </Col>
  </Row>
  <Row>
    <Col>
      <WidgetFooter />
    </Col>
  </Row>
</Container>
`;

exports[`DashboardLayout not collapsed sidebar not showing snapshot 1`] = `
<Container
  fluid={true}
  size="xl"
>
  <Row>
    <Col
      className="course-list-column"
      lg={
        Object {
          "offset": 0,
          "span": 12,
        }
      }
      xl={
        Object {
          "offset": 0,
          "span": 12,
        }
      }
    >
      test-children
    </Col>
    <Col
      className="sidebar-column"
      lg={
        Object {
          "offset": 0,
          "span": 12,
        }
      }
      xl={
        Object {
          "offset": 0,
          "span": 4,
        }
      }
    >
      <h2
        className="course-list-title"
      >
         
      </h2>
      <mockConstructor
        setSidebarShowing={[MockFunction hooks.setSidebarShowing]}
      />
    </Col>
  </Row>
  <Row>
    <Col>
      <WidgetFooter />
    </Col>
  </Row>
</Container>
`;

exports[`DashboardLayout not collapsed sidebar showing snapshot 1`] = `
<Container
  fluid={true}
  size="xl"
>
  <Row>
    <Col
      className="course-list-column"
      lg={
        Object {
          "offset": 0,
          "span": 12,
        }
      }
      xl={
        Object {
          "offset": 0,
          "span": 8,
        }
      }
    >
      test-children
    </Col>
    <Col
      className="sidebar-column"
      lg={
        Object {
          "offset": 0,
          "span": 12,
        }
      }
      xl={
        Object {
          "offset": 0,
          "span": 4,
        }
      }
    >
      <h2
        className="course-list-title"
      >
         
      </h2>
      <mockConstructor
        setSidebarShowing={[MockFunction hooks.setSidebarShowing]}
      />
    </Col>
  </Row>
  <Row>
    <Col>
      <WidgetFooter />
    </Col>
  </Row>
</Container>
`;
